### ISSUE TITLE
Accordion Header design pattern

### ISSUE THREAD

**Issue by laliq** [equinor/design-system#2091]
An example Introduced by EDS ➡️  **header with button** :
It is consider  as a bad practice, with the current [Accordion Design Pattern](https://www.w3.org/TR/wai-aria-practices-1.2/#accordion), and [Deque description](https://dequeuniversity.com/rules/axe/4.2/nested-interactive?application=axeAPI): 

> Interactive controls must not have focusable descendants .

Further more, this issue might need more attention because of the design pattern in EDS that is introduced today.

Lastly i would like to point some issues/questions that comes with EDS pattern:

- What is header and what is it's main function?
- What about creative approach  and design that engage users?
- What about accessibility?

**Comment by mimarz** [comment]
This also throws errors in development so suggest fixing this soon.

![image](https://user-images.githubusercontent.com/1070981/167618234-574b225c-a68b-4c48-910a-125eca6dbc20.png)

**Comment by akvam** [comment]
Backlog meeting: decided that this should be done.
